### PR TITLE
Plugin editor types

### PR DESCRIPTION
**Pull request by mariusandra**
## Changes

- Adds `@posthog/plugin-scaffold` types into the plugin source editor for sweet TypeScript support
- Changes the default plugin to one that encourages discovery (loosely based on @Twixes 's [advanced plugin kit](https://github.com/PostHog/posthog-plugin-advanced-kit/blob/a954ce9cf06274ea6525dc6d7102948072ffb763/src/index.ts#L18))
- This is just the first step though. We should add types for all other exported packages (e.g. aws sdk, plugin contrib, ...) as well.

## Checklist

- [ ] All querysets/queries filter by Organization, by Team, and by User
- [ ] Django backend tests
- [ ] Jest frontend tests
- [ ] Cypress end-to-end tests
- [ ] Migrations are safe to run at scale (e.g. PostHog Cloud) – present proof if not obvious

### CODE REVIEW

File: frontend/src/scenes/plugins/edit/PluginSource.tsx [hunk 1]
```diff
@@ -2,30 +2,31 @@ import React, { useEffect } from 'react'
...
-    }
+import SCAFFOLD_index from '!raw-loader!@posthog/plugin-scaffold/dist/index.d.ts'
+import SCAFFOLD_errors from '!raw-loader!@posthog/plugin-scaffold/dist/errors.d.ts'
+import SCAFFOLD_types from '!raw-loader!@posthog/plugin-scaffold/dist/types.d.ts'
+
+const defaultSource = `// Learn more about plugins at: https://posthog.com/docs/plugins/overview
+import { Plugin } from '@posthog/plugin-scaffold'
+
+type MyPluginType = Plugin<{
+  config: {},
```
Comment (Twixes) [inline frontend/src/scenes/plugins/edit/PluginSource.tsx:16]: This should match `defaultConfig`.

**Review by Twixes — CHANGES_REQUESTED** [review]
There is a problem in that this doesn't build

**Comment by mariusandra** [comment]
So, feedback addressed and it builds now. Still some issues with this PR. Not sure if they are blockers or not:

- We only have types for anything imported from `@posthog/plugin-scaffold`. No types for `fetch` or any of the other packages like `aws-sdk` here.
- As an improvement, I'd suggest adding two things to plugin-scaffold: 1) all pinned versions of all packages that plugin authors can import from into `devDependencies`, a build script that copies the types of all those packages into the plugin scaffold package for exporting. The `posthog` would just take these types from inside scaffold, instead of requiring us to install `aws-sdk` and other large things that slow down builds.
- We need to make sure to keep posthog, plugin-server and plugin-scaffold versions in sync. Automatic github actions could help.

**Review by Twixes — APPROVED** [review]
Nice.
As for improvements, I'd say nothing blocking. May be worth creating issue for some e.g. ensuring `plugin-scaffold` is up to date.

**Pull request merged**